storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+
+  // Min and Max values should be valid at this point.
+  EXPECT_TRUE(entry4->max_entry_ref.valid);
+  EXPECT_TRUE(entry4->min_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_EQ(min_id, entry4->min_entry_ref.entry_ref_tuple);
+  EXPECT_EQ(max_id, entry4->max_entry_ref.entry_ref_tuple);
+  EXPECT_TRUE(sma_test::chars_equal(
+      entry4->min_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry4->min_entry_ref.entry_ref_tuple, 4)));
+
+  EXPECT_TRUE(sma_test::chars_equal(
+      entry4->max_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry4->max_entry_ref.entry_ref_tuple, 4)));
+
+  // Test attribute 5, an 80 character char.
+  SMAEntry* entry5 = getEntryForAttribute(5);
+  EXPECT_EQ(kChar, entry5->type_id);
+  EXPECT_EQ(5, entry5->attribute);
+
+  // Min and Max values should be valid at this point.
+  EXPECT_TRUE(entry5->max_entry_ref.valid);
+  EXPECT_TRUE(entry5->min_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_EQ(min_id, entry5->min_entry_ref.entry_ref_tuple);
+  EXPECT_EQ(max_id, entry5->max_entry_ref.entry_ref_tuple);
+  EXPECT_TRUE(sma_test::chars_equal(
+      entry5->min_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry5->min_entry_ref.entry_ref_tuple, 5)));
+
+  EXPECT_TRUE(sma_test::chars_equal(
+      entry5->max_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry5->max_entry_ref.entry_ref_tuple, 5)));
+
+  // Test attribute 6, an 8 character varchar.
+  SMAEntry* entry6 = getEntryForAttribute(6);
+  EXPECT_EQ(kVarChar, entry6->type_id);
+  EXPECT_EQ(6, entry6->attribute);
+
+  // Min and Max values should be valid at this point.
+  EXPECT_TRUE(entry6->max_entry_ref.valid);
+  EXPECT_TRUE(entry6->min_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_EQ(min_id, entry6->min_entry_ref.entry_ref_tuple);
+  EXPECT_EQ(max_id, entry6->max_entry_ref.entry_ref_tuple);
+  EXPECT_TRUE(sma_test::varchars_equal(
+      entry6->min_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry6->min_entry_ref.entry_ref_tuple, 6)));
+
+  EXPECT_TRUE(sma_test::varchars_equal(
+      entry6->max_entry_ref.value,
+      tuple_store_->getAttributeValueTyped(entry6->max_entry_ref.entry_ref_tuple, 6)));
+
+  // Try kicking the index out of memory and then getting the variable length
+  // attributes back.
+  // We must call the destructor before attempting to create an index over the same memory.
+  index_.reset(nullptr);
+  index_.reset(new SMAIndexSubBlock(*tuple_store_,
+                                    *index_description_,
+                                    false,
+                                    index_memory_.get(),
+                                    kIndexSubBlockSize));
+  ASSERT_FALSE(index_->requiresRebuild());
+
+  // Ensure that the attributes are as they were before the rebuild process.
+  EXPECT_EQ(kChar, getEntryForAttribute(4)->type_id);
+
+  SMAEntry* nentry6 = getEntryForAttribute(6);
```

Comment:
new entry6? I have forgotten which is probably good evidence for changing it. I renamed it to `rebuilt_entry6`.